extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts

```diff
@@ -0,0 +1,506 @@
+							title: loc.updatingInstance(this._postgresModel.info.name),
+							cancellable: false
+						},
+						async (_progress, _token): Promise<void> => {
+							//all
+							// azdata arc postgres server edit -n <server group name> -e '' -re
+							try {
+								await this._azdataApi.azdata.arc.postgres.server.edit(
+									this._postgresModel.info.name, { engineSettings: `'' -re` });
+							} catch (err) {
+								// If an error occurs while resetting the instance then re-enable the reset button since
+								// the edit wasn't successfully applied
+								this.resetButton!.enabled = true;
+								throw err;
+							}
+							await this._postgresModel.refresh();
+						}
+
+					);
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.refreshFailed(error));
+				}
+			}));
+
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: this.saveButton },
+			{ component: this.discardButton },
+			{ component: this.resetButton }
+		]).component();
+	}
+
+	private initializeConnectButton() {
+
+		this.connectToServerButton = this.modelView.modelBuilder.button().withProps({
+			label: loc.connectToServer,
+			enabled: false,
+			CSSStyles: { 'max-width': '125px' }
+		}).component();
+
+		this.disposables.push(
+			this.connectToServerButton!.onDidClick(async () => {
+				this.connectToServerButton!.enabled = false;
+				if (!vscode.extensions.getExtension('microsoft.azuredatastudio-postgresql')) {
+					const response = await vscode.window.showErrorMessage(loc.missingExtension('PostgreSQL'), loc.yes, loc.no);
+					if (response !== loc.yes) {
+						this.connectToServerButton!.enabled = true;
+						return;
+					}
+
+					await vscode.commands.executeCommand('workbench.extensions.installExtension', 'microsoft.azuredatastudio-postgresql');
+				}
+
+				await this._postgresModel.getEngineSettings().catch(err => {
+					// If an error occurs show a message so the user knows something failed but still
+					// fire the event so callers can know to update (e.g. so dashboards don't show the
+					// loading icon forever)
+					if (err instanceof UserCancelledError) {
+						vscode.window.showWarningMessage(loc.pgConnectionRequired);
+					} else {
+						vscode.window.showErrorMessage(loc.fetchEngineSettingsFailed(this._postgresModel.info.name, err));
+					}
+					this._postgresModel.engineSettingsLastUpdated = new Date();
+					this._postgresModel._onEngineSettingsUpdated.fire(this._postgresModel._engineSettings);
+					this.connectToServerButton!.enabled = true;
+					throw err;
+				});
+
+				this.parameterContainer!.clearItems();
+				this.parameterContainer!.addItem(this.parametersTable);
+
+			}));
+	}
+
+	private initializeSearchBox() {
+		this.searchBox = this.modelView.modelBuilder.inputBox().withProps({
+			readOnly: false,
+			placeHolder: loc.searchToFilter
+		}).component();
+
+		this.disposables.push(
+			this.searchBox.onTextChanged(() => {
```

Comment:
So I think for this we're going to want to add in a debounce delay so if someone is typing really fast we don't have to do a ton of filters right in a row which would be a huge perf issue.

Don't have to handle it immediately but something to keep in mind as you continue to work on this.